### ISSUE TITLE
Improper Shader Subclass handling

### ISSUE THREAD

**Issue by JPMeehan** [foundryvtt/foundryvtt#10581]
Running into a specific shader typing issue while working on the Types project.
1. `RenderedSource##createMesh(shaderCls)` constructs the mesh with `new PointSourceMesh(this.#geometry, shaderCls.create(), state)`
2. `shaderCls` is annotated as `typeof AdaptiveLightingShader`
3. `AdaptiveLightingShader extends AbstractBaseShader extends BaseShaderMixin(PIXI.Shader)`
4. `PointSourceMesh` extends `PIXI.Mesh`, and the second argument in *that* constructor is expected to be `PIXI.MeshMaterial`
5. `PIXI.MeshMaterial extends PIXI.Shader`, adding properties that are NOT present in the `AdaptiveLightingShader` inheritance chain.

TL;DR `PointSourceMesh` canonically expects a different subclass of `PIXI.Shader` than is provided by Foundry in actual use.

----

Separate from #10227 because this seems to be a deeper issue in class construction.

**Comment by aaclayton** [comment]
bumping to high priority *in case* this requies API changes. @Feu-Secret to take a quick look. If not we can downgrade it back to the testing phase.
